File: frontend/types/project.ts
/**
 * 项目相关类型定义
 * 与后端 ProjectResponse 对齐
 */

export interface AgencySummary {
  id: string
  name: string
  logo?: string | null
}

export interface ProjectResponse {
  id: string
  name: string
  description?: string | null
  platform?: string | null
  brand_id?: string | null
  brand_name?: string | null
  client_display_name?: string | null
  brand_display_name?: string | null
  project_remark?: string | null
  status: string
  start_date?: string | null
  deadline?: string | null
  agencies: AgencySummary[]
  task_count: number
  created_at: string
  updated_at: string
}

export interface ProjectListResponse {
  items: ProjectResponse[]
  total: number
  page: number
  page_size: number
}

export interface ProjectCreateRequest {
  name: string
  description?: string
  platform?: string
  start_date?: string
  deadline?: string
  agency_ids?: string[]
  client_display_name?: string
  brand_display_name?: string
  project_remark?: string
}

export interface ProjectUpdateRequest {
  name?: string
  description?: string
  platform?: string
  start_date?: string
  deadline?: string
  status?: 'active' | 'completed' | 'archived'
  client_display_name?: string
  brand_display_name?: string
  project_remark?: string
}

File: backend/app/schemas/project.py
"""
项目相关 Schema
"""
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


# ===== 请求 =====

class ProjectCreateRequest(BaseModel):
    """创建项目请求（品牌方操作）"""
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    platform: Optional[str] = None
    start_date: Optional[datetime] = None
    deadline: Optional[datetime] = None
    agency_ids: Optional[List[str]] = None  # 分配的代理商 ID 列表
    client_display_name: Optional[str] = None
    brand_display_name: Optional[str] = None
    project_remark: Optional[str] = None


class ProjectUpdateRequest(BaseModel):
    """更新项目请求"""
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    platform: Optional[str] = None
    start_date: Optional[datetime] = None
    deadline: Optional[datetime] = None
    status: Optional[str] = Field(None, pattern="^(active|completed|archived)$")
    client_display_name: Optional[str] = None
    brand_display_name: Optional[str] = None
    project_remark: Optional[str] = None


class ProjectAssignAgencyRequest(BaseModel):
    """分配代理商到项目"""
    agency_ids: List[str]


# ===== 响应 =====

class AgencySummary(BaseModel):
    """代理商摘要"""
    id: str
    name: str
    logo: Optional[str] = None


class ProjectResponse(BaseModel):
    """项目响应"""
    id: str
    name: str
    description: Optional[str] = None
    platform: Optional[str] = None
    brand_id: Optional[str] = None
    brand_name: Optional[str] = None
    client_display_name: Optional[str] = None
    brand_display_name: Optional[str] = None
    project_remark: Optional[str] = None
    status: str
    start_date: Optional[datetime] = None
    deadline: Optional[datetime] = None
    agencies: List[AgencySummary] = []
    task_count: int = 0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    """项目列表响应"""
    items: List[ProjectResponse]
    total: int
    page: int
    page_size: int
